flow/render_status.py

```diff
@@ -0,0 +1,220 @@
+# make separate python class for render_status
+from .util import mistune
+from .scheduling.base import JobStatus
+
+
+class renderer:
+    """A class for rendering status in different format.
+
+    This class provides method and string output for rendering status output in different format,
+    currently supports: terminal, markdown and html
+
+    """
+
+    def __init__(self):
+        self.markdown_output = None
+        self.terminal_output = None
+        self.html_output = None
+
+    def generate_terminal_output(self):
+        self.terminal_output = mistune.terminal(self.markdown_output)
+
+    def generate_html_output(self):
+        self.html_output = mistune.html(self.markdown_output)
+
+    def render(self, template, template_environment, context, file, detailed, expand,
+               unroll, compact, pretty, option):
+        """Render the status in different format for print_status.
+
+        :param template:
+            User provided Jinja2 template file.
+        :type template:
+            str
+        :param template_environment:
+            template environment
+        :type template_environment:
+            :class:'FlowProject._template_environment'
+        :param context:
+            Context that includes all the information for rendering status output.
+        :type context:
+            dict
+        :param file:
+            Redirect all output to this file, defaults to sys.stdout.
+        :type file:
+            str
+        :param detailed:
+            Print a detailed status of each job.
+        :type detailed:
+            bool
+        :param expand:
+            Present labels and operations in two separate tables.
+        :type expand:
+            bool
+        :param unroll:
+            Separate columns for jobs and the corresponding operations.
+        :type unroll:
+            bool
+        :param compact:
+            Print a compact version of the output.
+        :type compact:
+            bool
+        :param pretty:
+            Prettify the output.
+        :type pretty:
+            bool
+        :param option:
+            Options for rendering format, currently supports:
+            'terminal'(default), 'markdown' and 'html'.
+        :type option:
+            str
+        """
+
+        # use Jinja2 template for status output
+        if template is None:
+            if detailed and expand:
+                template = 'status_expand.jinja'
+            elif detailed and not unroll:
+                template = 'status_stack.jinja'
+            elif detailed and compact:
+                template = 'status_compact.jinja'
+            else:
+                template = 'status.jinja'
+
+        def draw_progressbar(value, total, escape='', width=40):
```

Comment:
Should this be using `tqdm`? We will have it as an official dependency when #247 is merged.

Reply:
the use case here is just a static progress bar visualization and I am not sure if tqdm is need as I thought it is used to generate a dynamic progress bar, might be a overkill?

Reply:
If we chose to use tqdm, you can do something like the following in 2 lines:
```python
from tqdm import tqdm
def draw_progressbar(value, total, escape='', width=40):
    bar_format = escape + '|{bar}' + escape + '| {percentage:0.2f}%'
    return tqdm.format_meter(n=value, total=total, elapsed=0, ncols=width, bar_format=bar_format)
```

Example:
```python
>>> draw_progressbar(value=17, total=58)
'|█████████                      | 29.31%'
```
The advantages of this solution: slightly easier to read, ensures consistency of look and feel across signac-flow, uses Unicode characters for partially filled blocks.

Reply:
Just thought it was worth suggesting -- please feel free to accept or reject this suggestion (though I think it might be nice).

Reply:
Oh, one more thing -- the ratio (percentage) is included in the width here. So the characters in the percentage count towards the total width.

Reply:
thanks for the code example! I switched it to use tqdm.

Reply:
one minor issue, i try to keep consistent with the current format style and used the {bar:40} setting to make all the bar aligned regardless of the percentage value length, which turn out to be a new feature only available since this release: https://github.com/tqdm/tqdm/releases/tag/v4.35.0 and is newer then the #247 reqested ('tqdm>=4.10.0'). Would it be ok to just pump the minimum version or is there other constraints?

Reply:
@zhou-pj I think that would be fine to bump the version. Another alternative might be to keep our version at 4.10.0, use `ncols` to render _only_ the bar, and keep using format strings for the percentage and just concatenate the two strings.) That would be more similar to the old code, but just replacing the bar rendering pieces with tqdm.

Reply:
@zhou-pj Please feel free to bump the version number.

Reply:
@vyasr